Content.Client/VendingMachines/UI/VendingMachineMenu.xaml.cs

```diff
@@ -36,6 +41,22 @@ public VendingMachineMenu()
             };
         }
 
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+
+            // Don't clean up dummies during prediction or we'll just have to spawn them again
+            if (!disposing || !_timing.IsFirstTimePredicted)
+                return;
```

Comment:
Doesn't disposing handle this? It shouldn't be getting double-disposed.

Reply:
While testing, I had it debug log every entity it deleted, and it was spamming the terminal like crazy whenever I opened the UI.

Eventually I figured out that prediction was causing it to create and dispose the window each tick, so it was spawning and deleting the clientside entities each tick too. This logic makes it only spawn them once and keeps them around until the window is actually closed by the user.

Reply:
Thinking about this more, it might have the possibility of leaking clientside entities if it mispredicts opening. Should I just remove this and let it spawn and delete the client ents during prediction?